expression/expression.go

```diff
@@ -1161,6 +1161,13 @@ func canScalarFuncPushDown(scalarFunc *ScalarFunction, pc PbConverter, storeType
 
 	// Check whether this function can be pushed.
 	if !canFuncBePushed(scalarFunc, storeType) {
+		if pc.sc.InExplainStmt {
+			storageName := storeType.Name()
+			if storeType == kv.UnSpecified {
+				storageName = "storage layer"
+			}
+			pc.sc.AppendWarning(errors.New("Scalar function '" + scalarFunc.FuncName.L + "'(signature: " + scalarFunc.Function.PbCode().String() + ") can not be pushed to " + storageName))
```

Comment:
Looks like each time only one function can be shown. For a real world case it will be tricky.

Reply:
In TiDB implementation, for each `DNF` expression, If a function can not be pushed down, then all the other functions in the same `DNF` expression will not be checked, as far as I known, TiDB will convert filter expression into a list of `DNF` before predicate pushdown, so if `function_1` and `function_2` is not supported by the storage layer, for query:
`select * from a where function_1 or function_2`, only function_1 will be recorded.
but for query
`select * from a where function_1 and function_2`, both function_1 and function_2 will be recorded.

Reply:
why not reading the explain result to find which functions are not pushed down?